src/App.tsx

```diff
@@ -1,14 +1,48 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [active, setActive] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setTodos(data);
+        setInitialTodos(data);
+      })
+      .finally(() => setLoading(false));
```

Comment:
The `getTodos` API call lacks error handling. Consider adding a `.catch` block to handle any potential errors during the data fetching process. This will ensure the application can gracefully manage any issues that arise.